py-polars/polars/utils/udfs.py

```diff
@@ -450,6 +467,12 @@ def _expr(cls, value: StackEntry, col: str, param_name: str, depth: int) -> str:
                         if " " in e1
                         else f"{not_}{e1}.is_in({e2})"
                     )
+                elif op == "map_dict":
+                    if not self._caller_variables:
```

Comment:
nice! so we avoid recomputing the same thing?

Reply:
Yup :)